Uploader.m

```diff
@@ -24,85 +24,25 @@ - (void)uploadFiles:(RNFSUploadParams*)params
-  NSMutableData* reqBody = [NSMutableData data];
-
-  // add fields
-  for (NSString *key in _params.fields) {
-    id val = [_params.fields objectForKey:key];
-    if ([val respondsToSelector:@selector(stringValue)]) {
-      val = [val stringValue];
-    }
-    if (![val isKindOfClass:[NSString class]]) {
-      continue;
-    }
-
-    [reqBody appendData:formBoundaryData];
-    [reqBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"\r\n\r\n", key] dataUsingEncoding:NSUTF8StringEncoding]];
-    [reqBody appendData:[val dataUsingEncoding:NSUTF8StringEncoding]];
-    [reqBody appendData:[@"\r\n" dataUsingEncoding:NSUTF8StringEncoding]];
-  }
 
-  // add files
-  for (NSDictionary *file in _params.files) {
-    NSString *name = file[@"name"];
-    NSString *filename = file[@"filename"];
-    NSString *filepath = file[@"filepath"];
-    NSString *filetype = file[@"filetype"];
-
-    // Check if file exists
-    NSFileManager *fileManager = [NSFileManager defaultManager];
-    if (![fileManager fileExistsAtPath:filepath]){
-      // NSError* error = [NSError errorWithDomain:@"Uploader" code:NSURLErrorFileDoesNotExist userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat: @"Failed to open target file at path: %@", filepath]}];
-      // return _params.errorCallback(error);
-      NSLog(@"Failed to open target file at path: %@", filepath);
-      continue;
-    }
-
-    NSData *fileData = [NSData dataWithContentsOfFile:filepath];
-
-    [reqBody appendData:formBoundaryData];
-    [reqBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"; filename=\"%@\"\r\n", name.length ? name : filename, filename] dataUsingEncoding:NSUTF8StringEncoding]];
-
-    if (filetype) {
-      [reqBody appendData:[[NSString stringWithFormat:@"Content-Type: %@\r\n", filetype] dataUsingEncoding:NSUTF8StringEncoding]];
-    } else {
-      [reqBody appendData:[[NSString stringWithFormat:@"Content-Type: %@\r\n", [self mimeTypeForPath:filename]] dataUsingEncoding:NSUTF8StringEncoding]];
-    }
-
-    [reqBody appendData:[[NSString stringWithFormat:@"Content-Length: %ld\r\n\r\n", (long)[fileData length]] dataUsingEncoding:NSUTF8StringEncoding]];
-    [reqBody appendData:fileData];
-    [reqBody appendData:[@"\r\n" dataUsingEncoding:NSUTF8StringEncoding]];
+  // add file
+  NSDictionary *file = [_params.files objectAtIndex:0];
+  NSString *filepath = file[@"filepath"];
+    
+  NSFileManager *fileManager = [NSFileManager defaultManager];
+  if (![fileManager fileExistsAtPath:filepath]){
+    NSLog(@"Failed to open target file at path: %@", filepath);
+    NSError* error = [NSError errorWithDomain:@"Uploader" code:NSURLErrorFileDoesNotExist userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat: @"Failed to open target file at path: %@", filepath]}];
+    return _params.errorCallback(error);
```

Comment:
it's commented out because this code doesn't work if the file url/path is rooted at file:///